src/model/Model.cpp

```diff
@@ -1548,18 +1554,24 @@ namespace model {
     space1->setThermalZone(thermalZone);
     space1->setBuildingStory(buildingStory);
 
-    ModelObject clone = space1->clone(model);
```

Comment:
Oh I see, this probably wasn't deterministic because clone isn't deterministic.  You could make clone sort the children to fix that:

https://github.com/NREL/OpenStudio/blob/develop/src/model/ParentObject.cpp#L106

Reply:
See my above comment - would sorting fix the issue I was addressing? Namlelt that creating the export model twice did not produce identical models?

Reply:
Yes sorting the result of children in ParentObject_Impl::clone should go a long way towards making the models identical.  The objects will have different UUIDs so the models won't be completely identical.

Reply:
I don't know much about the clone itself but that seems to imply that children() doesn't return a specific order of children each time it is called? Its not just that they are coming back in a different order each time clone is called the surfaces are different, for example the first time clone is called I get this

![image](https://user-images.githubusercontent.com/21084412/151242970-227e9248-5b47-4305-aafc-2531df03b4d2.png)

and the second time I get this

![image](https://user-images.githubusercontent.com/21084412/151243020-f25e4afe-1ae7-4d31-807e-b1440077924c.png)

(screenshots are comparing the two xml files)

Reply:
I think my best course of action is to create my own simpler variant of exampleModel specifically for my test